src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -155,6 +151,14 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
                 _textChangeQueue.AddWork((oldDocument, newDocument));
         }
 
+        // Check if we're currently paused.  If so ignore this notification.  We don't want to any work in response
+        // to whatever the workspace is doing.
+        lock (_gate)
+        {
+            if (_isSynchronizeWorkspacePaused)
+                return;
+        }
+
         _synchronizeWorkspaceQueue.AddWork();
```

Comment:
>_synchronizeWorkspaceQueue [](http://example.com/codeflow?start=8&length=26)

This didn't change so I'm sure it's fine, but the Resume/Pause methods execute the calls onto the _synchronizeWorkspaceQueue from within the lock.